### PR TITLE
Add support for deCONZ ConBee

### PR DESCRIPTION
**Pull request by dhylands**
This factors the xbee specifi knowledge out of the existing stack.
zb-driver.js now contains common code for the driver level and
deconz-drive.js and xbee-driver.js contain the zigbee dongle
specific code.

This code also relies on deconz-api, a new repository created for
the deConz dongle which offers a similar API level to the xbee-api.

The ZDO code was also factored out into a separate repository.

### CODE REVIEW

File: zb-adapter.js [hunk 1]
```diff
@@ -1668,24 +1030,23 @@ class ZigbeeAdapter extends Adapter {
...
 
     const permitJoinFrame = this.zdo.makeFrame({
-      destination64: '000000000000ffff',
-      destination16: 'fffe',
+      destination64: 'ffffffffffffffff',
+      destination16: BROADCAST_ADDR.ROUTERS,
       clusterId: zdo.CLUSTER_ID.MANAGEMENT_PERMIT_JOIN_REQUEST,
       permitDuration: seconds,
-      trustCenterSignificance: 0,
+      trustCenterSignificance: 1,
```
[the remark this answers is not in the document]
Comment (dhylands) [reply zb-adapter.js]: Apparently this field is treated as though it was set to 1 anyways so I set it to be 1 to be consistent with the specs.

The spec says "This field shall always have a value of 1, indicating a request to change the Trust Center policy. If a frame is received with a value of 0, it shall be treated as having a value of 1."

**Comment by mrstegeman** [comment]
@dhylands There are 3 more outstanding comments above.

**Comment by dhylands** [comment]
> @dhylands There are 3 more outstanding comments above.

I only see the one about the if blocks - thanks for pointing that out.

**Comment by dhylands** [comment]
Never mind - I see there was a load more button that I missed

**Comment by dhylands** [comment]
OK - addressed the 3 remaining comments that I missed the first time around.

File: xbee-driver.js [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+  configureApiModeIfNeeded() {
+    if (DEBUG_flow) {
+      console.log('configureApiModeIfNeeded');
+    }
+    const configCommands = [];
+    if (this.apiMode != 1) {
+      configCommands.push(this.AT(AT_CMD.API_MODE,
+                                  {apiMode: 1}));
+      configCommands.push(this.AT(AT_CMD.API_MODE));
+    }
```
[the remark this answers is not in the document]
Comment (dhylands) [reply xbee-driver.js]: Done

File: deconz-driver.js [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+      new Command(SEND_FRAME, writeParamFrame),
+      new Command(WAIT_FRAME, {
+        type: C.FRAME_TYPE.WRITE_PARAMETER,
+        paramId: paramId,
+      }),
+    ];
+  }
+}
+
+const fh = DeconzDriver.frameHandler = {};
```
[the remark this answers is not in the document]
Comment (dhylands) [reply deconz-driver.js]: Done

File: index.js [hunk 4]
```diff
@@ -10,73 +10,88 @@
...
-}
+const SerialProber = require('serial-prober');
+
+const XBEE_FTDI_FILTER = {
+  // Devices like the UartSBee, use a generic FTDI chip and with
+  // an XBee S2 programmed with the right firmware can act like
+  // a Zigbee coordinator.
+  vendorId: '0403',
+  productId: '6001',
+  manufacturer: 'Digi',
```
[the remark this answers is not in the document]
Comment (dhylands) [reply index.js]: Yep - good ole copy/paste....

**Pull request merged**